iroh-net/src/magicsock/node_map/node_state.rs

```diff
@@ -222,17 +224,26 @@ impl NodeState {
             .udp_paths
             .paths
             .iter()
-            .map(|(addr, path_state)| DirectAddrInfo {
-                addr: SocketAddr::from(*addr),
-                latency: path_state.recent_pong.as_ref().map(|pong| pong.latency),
-                last_control: path_state.last_control_msg(now),
-                last_payload: path_state
-                    .last_payload_msg
-                    .as_ref()
-                    .map(|instant| now.duration_since(*instant)),
-                last_alive: path_state
-                    .last_alive()
-                    .map(|instant| now.duration_since(instant)),
+            .map(|(addr, path_state)| {
+                let mut sources: Vec<(Source, Duration)> = path_state
+                    .sources
+                    .iter()
+                    .map(|(source, instant)| (source.clone(), now.duration_since(*instant)))
+                    .collect();
+                sources.sort_by(|a, b| b.1.cmp(&a.1));
```

Comment:
this is such a small thing to complain, but here we go:

I assume the sources are sorted because it's nicer to show them that way in the cli. However, this line of code belongs to the hot path, so I argue this should not happen here. The information to do this is already returned and the caller can sort them if they want, in their most-likely not-hot path. In general this order is not used, since it's used for example for filtering. 

I won't die on this hill, but I'd like to hear opinions on the matter

Reply:
removed!